doc/guides/build/kconfig/setting.rst

```diff
@@ -167,6 +167,22 @@ The application configuration can come from the sources below. By default,
 6. Otherwise, :file:`prj.conf` is used if it exists in the application
    directory
 
+Per default all configuration files will be taken from the application's
+configuration directory except for files with an absolute path that are
+given with the ``CONF_FILE`` argument.
+
+The application's configuration directory will be defined as:
+
+1. If ``APPLICATION_CONFIG_DIR`` is specified by the user with
```

Comment:
I feel like this section belongs in doc/application/index.rst instead, no?

Reply:
I agree, it does.

But I also feel that this knowledge, and the knowledge regarding the prj.conf file belongs together.

Having those split can actually make it hard to understand exactly how the configuration / overlay scheme works.

Will take a second look, and if you have any good proposals, let me know.

Reply:
Why not put it in the application guide and link to it from here?